docs/naivebayes.md

```diff
@@ -0,0 +1,108 @@
+
+```
+predict_log_proba( X ) : Input a set of data and output the log probability of each sample under each distribution
+predict_proba( X ) : Input a set of data and output the probability of each sample under each distribution
+predict( X ) : Input a set of data and classify each sample under one of the distributions
+fit( X, y ) : Fit the Naive Bayes classifier to a set of training data, with X being the sample, and y being the correct classification for each sample
+```
+
+## Initialization
+
+Naive Bayes can be initialized a number of ways. The classifier can be initialized by either passing in the initial distribution objects or by passing in the constructor and the number of components.
+
+For example, here is an initialization using an already initialized normal distribution for class 0, a uniform distribution for class 1, and an exponential distribution for class 2 .
+
+```Python
+clf = NaiveBayes([ NormalDistribution( 5, 2 ), UniformDistribution( 0, 10 ), ExponentialDistribution( 1.0 ) ])
+```
+
+Since Naive Bayes classifiers simply compares the likelihood of a sample occurring under different models, it can be initialized with any model in pomegranate. This is assuming that all the models take the same type of input.
+
+```Python
+multi = MultivariateGaussianDistribution( means=[ 5, 5 ], covariance=[[ 2, 0 ], [ 0, 2 ]] )
+indie = IndependentComponentsDistribution( distributions=[ UniformDistribution(0, 10), UniformDistribution(0, 10) ])
+clf = NaiveBayes([ mutli, indie ])
+```
+
+Naive Bayes classifiers can also take Hidden Markov Models as input as well as Bayesian Networks.
+
+If all the models in the Naive Bayes classifier use the same type of model, then Naive Bayes can be initialized by passing in the constructor for the model and the number of classses there are.
+
+```Python
+clf = NaiveBayes( NormalDistribution, n_components=5 )
+```
+
+**Warning!** keep in mind that if Naive Bayes is initialized this way then all the classes must be fitted to some data before any prediction methods are called.
+
+Finally, Naive Bayes must be given at least a model or n_components must be specified otherwise a ValueError will be thrown.
+
+## Prediction
+
+Naive Bayes has three different prediction methods which all take the same input, an numpy array of samples. These methods are predict_proba, predict_log_proba, and predict which output, respectively, the probability of each sample occurring under each model, the log probability of each sample occurring under each model, and the model each sample is classified as.
+
+Since Naive Bayes uses Bayes Theorem, the probability of a sample occurring under each model is in respect to the total sample space of the sample occurring under all the models.
+
+Given our Naive Bayes classifier from above consisting of three univariate distribution, calling predict_proba on five samples would look like the following.
+
+```Python
+probs = clf.predict_proba( np.array([ 0, 1, 2, 3, 4 ]) )
```

Comment:
Show the classifier being initialized here, even if it's just copy/paste from before